ignite/contrib/handlers/time_profilers.py

```diff
@@ -0,0 +1,302 @@
+            ('mean', torch.mean(data).item()),
+            ('std', torch.std(data).item()),
+            ('total', torch.sum(data).item())
+        ])
+
+    def get_results(self):
+        events_to_ignore = [
+            Events.EXCEPTION_RAISED,
+            Events.GET_BATCH_COMPLETED,
+            Events.GET_BATCH_STARTED
+        ]
+        total_eh_time = sum([sum(self.event_handlers_times[e]) for e in Events if e not in events_to_ignore])
+        return OrderedDict([
+            ("processing_stats", self._compute_basic_stats(self.processing_times)),
+            ("dataflow_stats", self._compute_basic_stats(self.dataflow_times)),
+            ("event_handlers_stats",
+                dict([(str(e).replace(".", "_"), self._compute_basic_stats(self.event_handlers_times[e]))
+                      for e in Events if e not in events_to_ignore] + [("total_time", total_eh_time)])
+             ),
+            ("event_handlers_names", {str(e).replace(".", "_") + "_names": v
+                                      for e, v in self.event_handlers_names.items()})
+        ])
+
+    @staticmethod
+    def print_results(results):
+
+        def odict_to_str(d):
+            out = ""
+            for k, v in d.items():
+                out += "\t{}: {}\n".format(k, v)
+            return out
+
+        others = {k: odict_to_str(v) if isinstance(v, OrderedDict) else v
+                  for k, v in results['event_handlers_stats'].items()}
+
+        others.update(results['event_handlers_names'])
+
+        output_message = """
+--------------------------------------------
+- Time profiling results:
+--------------------------------------------
+
+Processing function time stats (in seconds):
+{processing_stats}
+
+Dataflow time stats (in seconds):
+{dataflow_stats}
+
+Time stats of event handlers (in seconds):
+- Total time spent:
+\t{total_time}
+
+- Events.STARTED:
+{Events_STARTED}
+Handlers names:
+{Events_STARTED_names}
+
+- Events.EPOCH_STARTED:
+{Events_EPOCH_STARTED}
+Handlers names:
+{Events_EPOCH_STARTED_names}
+
+- Events.ITERATION_STARTED:
+{Events_ITERATION_STARTED}
+Handlers names:
+{Events_ITERATION_STARTED_names}
+
+- Events.ITERATION_COMPLETED:
+{Events_ITERATION_COMPLETED}
+Handlers names:
+{Events_ITERATION_COMPLETED_names}
+
+- Events.EPOCH_COMPLETED:
+{Events_EPOCH_COMPLETED}
+Handlers names:
+{Events_EPOCH_COMPLETED_names}
+
+- Events.COMPLETED:
+{Events_COMPLETED}
+Handlers names:
+{Events_COMPLETED_names}
+
+""".format(processing_stats=odict_to_str(results['processing_stats']),
+           dataflow_stats=odict_to_str(results['dataflow_stats']),
+           **others)
+        print(output_message)
+        return output_message
+
+    @staticmethod
+    def write_results(output_path):
+        try:
+            import pandas as pd
+        except ImportError:
+            print("Need pandas to write results as files")
+            return
+
+        raise NotImplementedError("")
```

Comment:
Maybe we can also implement this if this make sense log times in a file.

Reply:
Hello,
Since the `get_results` method returns a nested OrderedDict, writing it into one csv file looks a bit tricky. We could either create separte files for `processing_stats`, `dataflow_stats`, etc or dump the results from the `print_results()` method into a text file (the output is already nicely formatted). Thoughts?

Reply:
Idea is to dump into csv non-aggregated times:

epoch | iteration | processing_stats | dataflow_stats | STARTED | EPOCH_STARTED | ITERATION_STARTED | ...
---|---|---|---|---|---|---|---
1 | 1 | 0.01 | 0.02 | 0.03 | 0.04 | 0.05
1 | 2 | 0.01 | 0.02 |  |  | 0.05
1 | 3 | 0.01 | 0.02 |  |  | 0.05

Something like that.

Reply:
Hi @vfdev-5

Reply:
Well, maybe we can do this in a single step (with a non-static method) without asking user to dump results. We can set into `__init__` the attributes
```
        self.dataflow_times = None
        self.processing_times = None
        self.event_handlers_times = None
```
such that user can reuse them freely. So, I would remove `dump_results` method.

Can you also post a screen shot of obtained csv as a table please.

Reply:
Yes, that sounds good. I pushed the changes.

Also, attaching the screenshot. Does this look good?
![file_dump](https://user-images.githubusercontent.com/5177126/73610250-f3e23a80-45dd-11ea-9e55-3bcd33899b73.png)

Reply:
Oh sorry, you wanted the screenshot as a table. PFA
![Screenshot 2020-02-02 at 17 08 37](https://user-images.githubusercontent.com/5177126/73610332-b336f100-45de-11ea-8c43-0d81fa4408d2.png)

Reply:
@ykumards table looks very cool :)